packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -27,6 +32,49 @@ class _FuchsiaLogReader extends DeviceLogReader {
   String toString() => name;
 }
 
+class FuchsiaDevices extends PollingDeviceDiscovery {
+  FuchsiaDevices() : super('Fuchsia devices');
+
+  @override
+  bool get supportsPlatform => true;
+
+  @override
+  bool get canListAnything => fuchsiaWorkflow.canListDevices;
+
+  @override
+  Future<List<Device>> pollingGetDevices() async {
+    if (!fuchsiaWorkflow.canListDevices) {
+      return <Device>[];
+    }
+    final String text = fuchsiaSdk.netls();
+    final List<FuchsiaDevice> devices = <FuchsiaDevice>[];
+    parseFuchsiaDeviceOutput(text, devices: devices);
+    return devices;
+  }
+
+  @override
+  Future<List<String>> getDiagnostics() async => const <String>[];
+}
+
+/// Parses output from the netls tool into fuchsia devices.
+///
+/// Example output:
+/// $ ./netls
+/// > device liliac-shore-only-last (fe80::82e4:da4d:fe81:227d/3)
+@visibleForTesting
+void parseFuchsiaDeviceOutput(String text, {@required List<FuchsiaDevice> devices}) {
```

Comment:
Have `[netls()]` run async and return a `Stream<Uint8List>` (and then maybe make this method an async generator that returns a `Stream<FuchsiaDevice>` instead of a `Future<List<FuchsiaDevice>>`).